### PR TITLE
Upgrade PySide2 for Windows to 5.15.2

### PR DESCRIPTION
**Pull request by junbo75**
Signed-off-by: Junbo Liang <68558268+junbo75@users.noreply.github.com>

- Upgraded Windows PySide2 package to the same version as the Linux package
- Built with O3DE version of OpenSSL
- Added build scripts for the Windows PySide2 package

Tested the package using AWS Resource Mapping Tool and the QtForPython gem.
image
image
image

### CODE REVIEW

File: package-system/pyside2-qt/build_config.json [hunk 1]
```diff
@@ -1,28 +1,45 @@
...
+    "package_url": "https://code.qt.io/cgit/pyside/pyside-setup.git",
+    "package_license": "custom",
+    "package_license_file": "pyside2/LICENSE.LGPLv3",
+    "patch_file": "pyside_o3de-v5.14.2.3.patch",
+    "cmake_find_target": "Findpyside2.cmake",
+    "Platforms": {
+        "Windows": {
+            "Windows": {
+                "depends_on_packages" :[ 
+                    ["python-3.7.10-rev1-windows", "851383addea5c54b7c6398860d04606c1053f1157c2edd801ec3d47394f73136", ""],
```
Comment (Jackie9527) [inline package-system/pyside2-qt/build_config.json:14]: https://github.com/o3de/o3de/blob/65f037278539237c7a5b01535c6e56cf72417ab5/cmake/LYPython.cmake#L35

Python has been upgraded to 3.7.12.

**Review by Jackie9527 — CHANGES_REQUESTED** [review]
Python has been upgraded to 3.7.12.

https://github.com/o3de/o3de/blob/65f037278539237c7a5b01535c6e56cf72417ab5/cmake/LYPython.cmake#L35

File: package-system/pyside2-qt/build_config.json (hunk 1 above)
Comment (junbo75) [reply package-system/pyside2-qt/build_config.json:14]: Please check the reply below. Thanks.

File: package-system/pyside2-qt/build_windows.cmd [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+REM 
+
+REM Prerequisites:
+REM - Visual Studio 2017 is installed
+REM - 7-zip is installed: https://www.7-zip.org/
+
+REM Download the prebuilt clang package from the Qt server. The current version of PySide2 is not compatible with the latest version of clang package.
+echo Downloading the prebuilt clang package from the Qt server
+call curl https://download.qt.io/development_releases/prebuilt/libclang/libclang-release_80-based-windows-vs2017_64.7z -L -o %TEMP_FOLDER%\libclang-release_80-based-windows-vs2017_64.7z
+"C:\Program Files\7-Zip\7z.exe" x %TEMP_FOLDER%\libclang-release_80-based-windows-vs2017_64.7z -o%TEMP_FOLDER%\libclang-release_80-based-windows-vs2017_64 -y
```
Comment (AMZN-puvvadar) [inline package-system/pyside2-qt/build_windows.cmd:16]: Consider building more resiliency around the 7zip path. i.e. turning it into a script parameter with this path as a default value in the event a user installs it somewhere else. Another one would be checking the path and printing a brutally clear error message on absence and exiting.

Comment (junbo75) [reply package-system/pyside2-qt/build_windows.cmd]: Good point. Just made the update to check the 7-zip path and allow users to change that via an environment variable.

**Review by AMZN-puvvadar — APPROVED** [review]

**Comment by junbo75** [comment]
Close this PR since @spham-amzn is also working on the PySide build scripts for both Windows and Mac. He will add OpenSSL support in his PR.

**Pull request closed without merging**